app/javascript/controllers/nextflow/samplesheet_controller.js

```diff
@@ -101,28 +91,47 @@ export default class extends Controller {
   // tracks filter state of search/clear buttons on filter
   #filterEnabled = false;
 
+  #queuedMetadataChanges = {};
+
+  #samplesheetReady = false;
+  #allowedToUpdateSamples;
+
   connect() {
-    if (this.hasWorkflowAttributesTarget) {
-      this.#setSamplesheetParametersAndData();
-      this.#updateMetadataColumnHeaderNames();
-      this.#disableProcessingState();
-    }
+    this.#updateMetadataColumnHeaderNames();
     this.element.setAttribute("data-controller-connected", "true");
   }
 
-  #setSamplesheetParametersAndData() {
-    this.#samplesheetProperties = JSON.parse(
-      this.samplesheetPropertiesTarget.innerHTML,
+  disconnect() {
+    this.samplesheetParamsFormTarget.removeEventListener(
```

Comment:
This will through an error when using automated workflows, since they never render the `this.samplesheetParamsFormTarget`.  Gaurd with `this.hasSamplesheetParamsFormTarget && this.boundAmendForm` before removing the listener